cdb2api/cdb2api.c

```diff
@@ -1180,11 +1180,13 @@ static int cdb2_socket_pool_get_ll(const char *typestr, int dbnum, int *port)
         }
     }
 
-    struct sockpool_msg_vers0 msg;
+    struct sockpool_msg_vers0 msg = {0};
     /* Please may I have a file descriptor */
-    bzero(&msg, sizeof(msg));
     msg.request = SOCKPOOL_REQUEST;
     msg.dbnum = dbnum;
+    if (strlen(typestr) >= sizeof(msg.typestr)) {
```

Comment:
Add the logic for donate too, otherwise it will lead to lot of idle appsock connections.

Reply:
Good thinking, thanks.